reconcile/openshift_serviceaccount_tokens.py

```diff
@@ -16,21 +15,33 @@
 QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
 
 
-def construct_sa_token_oc_resource(name, sa_name, sa_token):
+def construct_sa_token_oc_resource(name, sa_token):
     body = {
         "apiVersion": "v1",
         "kind": "Secret",
         "type": "Opaque",
         "metadata": {
             "name": name,
         },
-        "data": {"token": base64.b64encode(sa_token).decode("utf-8")},
+        "data": {"token": sa_token},
     }
     return OR(
         body, QONTRACT_INTEGRATION, QONTRACT_INTEGRATION_VERSION, error_details=name
     )
 
 
+def get_token_for_service_account(service_account, tokens):
+    result = []
+    for token in tokens:
+        if (
+            token["metadata"]["annotations"]["kubernetes.io/service-account.name"]
+            == service_account
+            and service_account + "-token" in token["metadata"]["name"]
```

Comment:
it would be better to filter on the secret `type` field than on a name suffix:
```
type: kubernetes.io/service-account-token
```

Reply:
+1 for what patrick said - that condition should be sufficient and we don't need to rely on name of the secret
i also wonder where a secret with the "-token" suffix comes from

Reply:
Ah thanks for point this out I see it now.